engine_details/development/debugging/using_cpp_profilers.rst

```diff
@@ -12,150 +12,31 @@ To this end, profilers are useful tools.
     but using C++ profiler may be useful in cases where the GDScript profiler
     is not accurate enough or is missing information due to bugs in the profiler.
 
-Recommended profilers
----------------------
+There are two main types of profilers: sampling profilers and tracing profilers.
 
-- `VerySleepy <http://www.codersnotes.com/sleepy/>`__ (Windows only)
-- `HotSpot <https://github.com/KDAB/hotspot>`__ (Linux only)
-- `Xcode Instruments <https://developer.apple.com/xcode/>`__ (macOS only)
+Sampling profilers periodically interrupt the running program and take a "sample",
+which records which functions are running. The profiler can aggregate this data to
+produce a graph which shows which functions the program spent the most time in.
 
-These profilers may not be the most powerful or flexible options, but their
-standalone operation and limited feature set tends to make them easier to use.
+Tracing profilers work by recording application-specific events over a short
+period of time (usually only a few seconds), producing a log called a "trace".
+The profiler can use the trace to produce a graph showing a high-level timeline of
+what happened, when and how long it took.
```

Comment:
To me, the most important features of tracing profilers are that they're accurate, and that they require manual setup.

```suggestion
Tracing profilers work by recording application-specific events (such as the start and end of a single frame), producing a log called a "trace".
The profiler can use the trace to produce a graph showing an accurate high-level timeline of
what happened.
```

Reply:
I've accepted your suggestion (thanks!), however...

> I've also recorded for plenty longer than a few seconds with tracy, without performance issues. I wouldn't be surprised if it supported tracing hours at a time 😅

We've barely added any events yet, though. Once something like https://github.com/godotengine/godot/pull/112707 is merged, this is going to change. In [their Godotfest presentation](https://www.youtube.com/watch?v=3BIteSVcO28), one of the "Halls of Torment" devs described how they implemented something similar, and could only run it for a few seconds because otherwise it would generate gigabytes of data.